pkg/auth/scope/publicshare.go

```diff
@@ -71,7 +71,8 @@ func publicshareScope(scope *authpb.Scope, resource interface{}) (bool, error) {
 }
 
 func checkStorageRef(s *link.PublicShare, r *provider.Reference) bool {
-	// r: <id:<storage_id:$storageID node_id:$nodeID path:$path > >
+	// TODO @ishank011 con you explain how this is used?
```

Comment:
For every incoming request of the above types, we extract the `Reference` object and go through all the objects that are in the current scope to see if access can be allowed to that particular reference. For example, we create a token with access to a particular public share and we store the resource ID that share points to.